model/models/llama/model.go

```diff
@@ -120,11 +120,17 @@ type Layer struct {
 	MLP           *MLP
 }
 
-func (l *Layer) Forward(ctx ml.Context, hiddenState, positionIDs ml.Tensor, cache kvcache.Cache, opts *Options) ml.Tensor {
+func (l *Layer) Forward(ctx ml.Context, hiddenState, positionIDs, outputs ml.Tensor, cache kvcache.Cache, opts *Options) ml.Tensor {
```

Comment:
```suggestion
// Forward processes input through a transformer layer
// Parameters:
//   - ctx: Context for tensor operations
//   - hiddenState: Input hidden state tensor 
//   - positionIDs: Position embeddings for attention
//   - outputs: If non-nil (in final layer), specifies which token positions need logits.
//             This enables pruning calculations for tokens we don't need, since in 
//             inference we often only need the logits for the most recent token.
//             Pruning here saves computation in MLP and remaining layers.
//   - cache: KV-cache for attention
//   - opts: Model options including epsilon for layer norm
func (l *Layer) Forward(ctx ml.Context, hiddenState, positionIDs, outputs ml.Tensor, cache kvcache.Cache, opts *Options) ml.Tensor {
```

I want to add some docs around what the outputs are, since it isn't super obvious without tracing the code

Reply:
Are you imagining that we would include this for every model (and every function in the model)? I agree that it's good to document but it seems a bit verbose, especially explaining the outputs logic everywhere.

Reply:
Actually, I agree with you, disregard this!